pkg/3scale/amp/component/apicast.go

```diff
@@ -830,6 +830,14 @@ func (apicast *Apicast) podAnnotations() map[string]string {
 		annotations[key] = val
 	}
 
+	for key, val := range apicast.Options.StagingPodTemplateAnnotations {
```

Comment:
This will apply production and staging annotations to both staging and production, I would reuse this function and call them something like prodPodAnnotations and staginPodAnnotations.

Reply:
done. Thank you!